netkan/netkan/scheduler.py

```diff
@@ -48,6 +49,52 @@ def schedule_all_netkans(self):
         for batch in sqs_batch_entries(messages):
             self.client.send_message_batch(**self.sqs_batch_attrs(batch))
 
+    def cpu_credits(self, cloudwatch, instance_id, start, end):
+        stats = cloudwatch.get_metric_statistics(
+            Dimensions=[{'Name': 'InstanceId', 'Value': instance_id}],
+            MetricName='CPUCreditBalance',
+            Namespace='AWS/EC2',
+            StartTime=start.strftime("%Y-%m-%dT%H:%MZ"),
+            EndTime=end.strftime("%Y-%m-%dT%H:%MZ"),
+            Period=10,
+            Statistics=['Average'],
+        )
+        # A pass consumes around 40 credits, with an accrue rate of 24/hr.
+        # So running every 2 hours should see using just a touch less than
+        # we gain in that time period.
+        creds = 0
+        try:
+            creds = stats['Datapoints'][0]['Average']
+        except IndexError:
+            logging.error("Couldn't acquire CPU Credit Stats")
+        return int(creds)
+
+    def volume_credits(self, cloudwatch, instance_id, start, end):
+        client = boto3.client('ec2')
```

Comment:
I think we could use `self.client` instead of getting a new one here.

Reply:
Because we only use the ec2 client once. We could use self.cloudwatch instead of passing it in though.

Reply:
Actually going to leave that for a later refactor. It's kinda hard to test without being on the instance, without doing a lot more work.